src/bp/ui-studio/src/web/components/App/index.tsx

```diff
@@ -37,7 +39,18 @@ class App extends Component {
     this.props.fetchUser()
   }
 
+  // Prevents re-rendering the whole layout when the user changes. Fixes a bunch of warnings & double queries
+  shouldComponentUpdate() {
+    return false
```

Comment:
props and state will never ever change?

Reply:
Not for that component. It renders routes & the general layout. Child components are connected to the store and it should not have a state. 

The main issue is that when it re-renders, the whole app is reloaded (previous components unmounted and remounted very quickly), but it does so while they are initializing and reloads twice. 

There could be other solutions (different refactors) but would require more tests, while the only issue here is the addition of 'user' as a props which is not even used in rendering

![image](https://user-images.githubusercontent.com/42552874/62179325-2737ca00-b31a-11e9-9858-1806a30fd484.png)